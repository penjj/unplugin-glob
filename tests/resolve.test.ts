import path from 'node:path'
import { describe, expect, test } from 'vitest'
import { rollup } from 'rollup'
import { build } from 'esbuild'
import glob from 'fast-glob'
import Esbuild from 'rollup-plugin-esbuild'
import RollupPlugin from '../src/rollup'
import EsBuildPlugin from '../src/esbuild'

describe('resolve', async () => {
  const fixtures = await glob(['./fixtures/*.{js,ts}', '!**/*.d.ts'], {
    absolute: true,
    cwd: __dirname,
  })
  for (const fixture of fixtures) {
    const ext = path.extname(fixture)
    const filename = path.basename(fixture, ext)

    test(`Rollup: ${filename}`, async () => {
      const bundle = await rollup({
        input: fixture,
        treeshake: false,
        plugins: [
          RollupPlugin({
            dts: path.resolve(path.dirname(fixture), `${filename}-glob`),
          }),
          Esbuild(),
        ],
      })
      const { output } = await bundle.generate({})
      expect(output[0].code).toMatchSnapshot()
    })

    test(`EsBuild: ${filename}`, async () => {
      const bundle = await build({
        entryPoints: [fixture],
        treeShaking: false,
        bundle: true,
        write: false,
        plugins: [
          EsBuildPlugin({
            dts: path.resolve(path.dirname(fixture), `${filename}-glob`),
          }) as any,
        ],
      })
      const { outputFiles } = bundle
      expect(outputFiles[0].text).toMatchSnapshot()
    })
  }
})
