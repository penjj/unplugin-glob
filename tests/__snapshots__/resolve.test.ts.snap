// Vitest Snapshot v1

exports[`resolve > basic 1`] = `
"const a = 'a';

const foo = 'bar';

export { a, foo };
"
`;

exports[`resolve > exclude-self 1`] = `
"const a = 'a';

const foo = 'bar';

export { a, foo };
"
`;

exports[`resolve > import-all 1`] = `
"const a = 'a';

const foo = 'bar';

console.log(a);
console.log(foo);

export { a, foo };
"
`;

exports[`resolve > ts 1`] = `
"const a = 'a';

const foo = 'bar';

export { a, foo };
"
`;
